File: lintcode/2_trailing-zeros/trailing-zeros.py
# coding:utf-8
'''
@Copyright:LintCode
@Author:   chenweikang
@Problem:  http://www.lintcode.com/problem/trailing-zeros
@Language: Python
@Datetime: 16-04-24 21:33
'''

class Solution:
    # @param n a integer
    # @return ans a integer
    def trailingZeros(self, n):
        cnt = 0
        while n:
            n /= 5
            cnt += n
        return cnt

File: lintcode/2_trailing-zeros/trailing-zeros.cpp
/*
@Copyright:LintCode
@Author:   chenweikang
@Problem:  http://www.lintcode.com/problem/trailing-zeros
@Language: C++
@Datetime: 16-04-24 21:32
*/

class Solution {
  public:
    // param n : description of n
    // return: description of return
    long long trailingZeros(long long n) {
        long long cnt = 0;
        while (n) {
            n /= 5;
            cnt += n;
        }
        return cnt;
    }
};
